### PR TITLE
Feature/dino score

### PR DESCRIPTION
**Pull request by begumcig**
## Description

This PR introduces the DINO Score metric.

The core implementation was contributed by @Prashankavi in PR #321 and this branch brings their work into the main PrunaAI repository. I made minor adjustments (styling, tests) to align with our current codebase and standards.

Full credit for the original implementation goes to @Prashankavi 💜💜💜 Thanks a lot for your amazing contribution! 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/pruna/evaluation/metrics/metric_dino_score.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    def __init__(self, device: str | torch.device | None = None, call_type: str = SINGLE):
+        super().__init__()
+        if device is not None and device not in self.runs_on:
+            pruna_logger.error(f"DinoScore: device {device} not supported. Supported devices: {self.runs_on}")
+            raise
+        self.device = set_to_best_available_device(device)
+        self.call_type = get_call_type_for_single_metric(call_type, self.default_call_type)
+        # Load the DINO ViT-S/16 model once
+        self.model = timm.create_model("vit_small_patch16_224.dino", pretrained=True)
+        self.model.eval().to(device)
```
[the remark this answers is not in the document]
Comment (begumcig) [reply src/pruna/evaluation/metrics/metric_dino_score.py]: thank you bugbot that's amazing

**Review by sdiazlor — APPROVED** [review]
LGTM :)
